File: ViperC-Quick.xctemplate/Objective-C/___FILEBASENAMEASIDENTIFIER___InteractorTests.m
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//  Copyright © ___YEAR___ ___ORGANIZATIONNAME___. All rights reserved.
//

#import <Quick/Quick.h>
#import <Expecta/Expecta.h>
#import "___VARIABLE_viperModuleName___Interactor.h"
#import "___VARIABLE_viperModuleName___ViewController.h"
#import "___VARIABLE_viperModuleName___Presenter.h"
#import "___VARIABLE_viperModuleName___Router.h"

QuickSpecBegin(___VARIABLE_viperModuleName___InteractorTests)

describe(@"___VARIABLE_viperModuleName___Interactor Tests", ^{
    __block ___VARIABLE_viperModuleName___ViewController *view;
    ___VARIABLE_viperModuleName___Interactor *interactor = [[___VARIABLE_viperModuleName___Interactor alloc] init];
    __block ___VARIABLE_viperModuleName___Presenter *presenter;

    beforeSuite(^{
        ___VARIABLE_viperModuleName___Router *router = [[___VARIABLE_viperModuleName___Router alloc] init];
        view = (___VARIABLE_viperModuleName___ViewController *) [___VARIABLE_viperModuleName___Router createModule];
        presenter = [[___VARIABLE_viperModuleName___Presenter alloc] initWithInterface:view
                                                                            interactor:interactor
                                                                                router:router];
        view.presenter = presenter;
        interactor.output = presenter;
    });
});


QuickSpecEnd

File: ViperC-Quick.xctemplate/Swift/___FILEBASENAMEASIDENTIFIER___InteractorTests.swift
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//  Copyright © ___YEAR___ ___ORGANIZATIONNAME___. All rights reserved.
//

import Quick
import Nimble

//@testable import YOUR_MODULE_NAME

class ___VARIABLE_viperModuleName___InteractorTests: QuickSpec {

    override func spec() {
        var view: ___VARIABLE_viperModuleName___ViewController!
        let interactor = ___VARIABLE_viperModuleName___Interactor()
        var presenter: ___VARIABLE_viperModuleName___Presenter!

        beforeSuite {
            let router = ___VARIABLE_viperModuleName___Router()
            view = ___VARIABLE_viperModuleName___Router.createModule() as? ___VARIABLE_viperModuleName___ViewController
            presenter = ___VARIABLE_viperModuleName___Presenter(interface: view,
                                                                interactor: interactor,
                                                                router: router)

            view.presenter = presenter
            interactor.presenter = presenter
        }
    }

}
